### DISCUSSION TITLE
Frequent disconnects via transport close

### DISCUSSION THREAD

**Discussion by vmorrisPolar** [miguelgrinberg/python-socketio#1401] [Q&A]
We are working on a web application that maintains a connection with a 3D printer via socketio. We are experiencing frequent disconnects, several times a minute. The client (Bambu X1C 3D printer) is running python-socketio.

The server is running "socket.io": "^4.3.1",

Connected at 02:56, disconnected 21 seconds later at 03:17, citing "transport close"
connected at 28:17, disconnected 21 seconds later at 28:38, citing "transport close"

Can anyone point me in the right direction towards improving the health and longevity of this socket?

![image](https://github.com/user-attachments/assets/4ae4e21c-a39d-4fb0-b483-6c2e5307eb98)

**Comment by miguelgrinberg** [comment]
Please do not write issues on my repositories when you need help with your own applications. The discussions forum is the right place.

Now regarding your problem, see the Troubleshooting section of the documentation to learn how to enable logging on your Python client. Hopefully the client-side logs will provide some context on the reason for the disconnections.

**Reply by vmorrisPolar** [reply]
Hey, really apologize about that. I'll turn on logs for the client and see what we can find out, then report back.
